src/main/java/org/spongepowered/api/service/scheduler/TaskBuilder.java

```diff
@@ -0,0 +1,144 @@
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.service.scheduler;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Represents a builder to create a {@link Task}.
+ */
+public interface TaskBuilder {
+
+    /**
+     * Sets whether the task should run asynchronous, outside of the main loop,
+     * and in it's own thread. By default, tasks are synchronous.
+     *
+     * <p>A synchronous task is ran in alignment with the game's main loop, in
+     * the same thread. Each synchronous task is ran in series with the tick
+     * cycle. It is safe to manipulate game data when running in this mode.</p>
+     *
+     * <p>In contrast, a task set to run asynchronously will run independently
+     * of the tick cycle and in it's own thread. Therefore the task is <b>not
+     * thread safe</b> with game data and care must be taken. It is strongly
+     * advised to <b>not</b> manipulate game data in asynchronous tasks.</p>
+     *
+     * <p>It is not possible to schedule a task in unit ticks when running
+     * asynchronously, that is why the returned builder is specialised for just
+     * building async tasks. The delay and interval are converted by the
+     * implementation to the equivalent wall clock time.</p>
+     *
+     * @return A specialised asynchronous task builder
+     */
+    TaskBuilder async();
+
+    /**
+     * Sets the {@link Runnable} to run when this task executes.
+     *
+     * @param runnable The actual task to run
+     * @return This builder, for chaining
+     */
+    TaskBuilder execute(Runnable runnable);
+
+    /**
+     * Sets the delay before the task runs. This delay is an initial offset,
+     * subsequent runs (when the interval is not 0) will not be offset. By
+     * default, the delay is 0
+     *
+     * @param delay The delay in the given {@link TimeUnit}
+     * @param unit The unit the delay is in
+     * @return This builder, for chaining
+     * @throws IllegalArgumentException If the delay is below 0
+     */
+    TaskBuilder delay(long delay, TimeUnit unit);
+
+    /**
+     * Sets the delay in ticks before the task runs. This only has an affect
+     * when the builder is in sync mode, if {@link #async()} has been called,
+     * this method does nothing.
```

Comment:
It's not a good idea to make this simply do nothing in one special case. It would be really more consistent especially with `Task.getDelay()` to make this use the default unit for the selected type of task (in our implementations most likely ms for async, ticks for sync).